File: core/src/test/scala/diplomacy/board/ProvinceSpec.scala
package diplomacy.board

import diplomacy.UnitSpec
import diplomacy.mock.board._

class ProvinceSpec extends UnitSpec {
  "A province" when {
    "it is a supply-center" should {
      "add * to a return-value of #toString." in {
        mar.toString should be("Mar*")
      }
    }
  }
}

File: test/board/province-spec.ts
import * as chai from "chai"

import { board } from "./../../src/board"
const { Name, Province } = board

const should = chai.should()

describe("Province", () => {
  describe("if the province is supply center", () => {
    it("adds *.", () => {
      const prov = new Province(new Name("Lvp"), "England", true)

      prov.toString().should.equal("Lvp*")

      should.equal(prov.homeOf, "England")
    })
  })
})
